### PR TITLE
wallet: Fix potential memory leak when loading a corrupted wallet file

### PR DESCRIPTION
**Pull request by practicalswift**
Fix potential memory leak when loading a corrupted wallet file.

### CODE REVIEW

File: src/wallet/wallet.cpp [hunk 1]
```diff
@@ -3886,15 +3886,12 @@ CWallet* CWallet::CreateWalletFromFile(const std::string walletFile)
         uiInterface.InitMessage(_("Zapping all transactions from wallet..."));
 
         std::unique_ptr<CWalletDBWrapper> dbw(new CWalletDBWrapper(&bitdb, walletFile));
-        CWallet *tempWallet = new CWallet(std::move(dbw));
+        std::unique_ptr<CWallet> tempWallet(new CWallet(std::move(dbw)));
```
Comment (eklitzke) [inline src/wallet/wallet.cpp]: nit: If you move the previous line into this one, you can omit the `std::move` operation.

**Comment by gmaxwell** [comment]
Please avoid titling PR's with things like "Fix memory leak" for things where there isn't an actual leak in practice. (e.g. because it exits).  Changes like this are good but they mess up people trying to figure out if an issue they are encounter has been fixed ("I have a memory leak, oh they just fixed one. They's probably it.").

**Review by gmaxwell — APPROVED** [review]
utACK

**Comment by practicalswift** [comment]
@gmaxwell That's a good point! I've now changed to "Fix potential memory leak" - works or still room for improvement? :-)

**Comment by MarcoFalke** [comment]
utACK c8b4b80df8346739c9a52113fd6673db1574e1f3

**Comment by practicalswift** [comment]
Rebased!

File: src/wallet/wallet.cpp (hunk 1 above)
Comment (jonasschnelli) [inline src/wallet/wallet.cpp]: I think @eklitzke is right here that it would be good to avoid that std::move.

Comment (promag) [inline src/wallet/wallet.cpp]: Too long?

**Pull request merged**